lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -188,7 +188,7 @@ def parse_relationship(assoc_name, assoc_data, options)
             end
 
           polymorphic = (options[:polymorphic] || []).include?(assoc_name.to_sym)
-          hash["#{prefix_key}_type".to_sym] = assoc_data['type'] if polymorphic
+          hash["#{prefix_key}_type".to_sym] = assoc_data['type'].singularize.capitalize if polymorphic
```

Comment:
could we `camelize` instead of `capitalize`? that would add support for snake cased multi word model names, right?